Add Prices tab to Products overview

mamhoff opened the pull request:
We have great functionality for dealing with multiple prices for each variant. This is an
absolutely basic interface so that as a store admin, it's possible to edit and update prices.

I'm unsure about allowing deletion and have left it out for now: When you delete a price,
you might end up with no price that has `is_default` set, which will then make the variant
in question become "unavailable". As a new price gets the `is_default` boolean anyways, the old
one becomes so meaningless it's as if it was deleted.

There are plans for a much better interface, but I need something simple that works for 1.3
to get out the door.

mamhoff commented:
Index view without deletion
![solidus administration prices 2016-05-06 17-33-44](https://cloud.githubusercontent.com/assets/703401/15086749/e0f5da02-13b0-11e6-9b14-80af1a789b5f.png)

Edit view (you can only edit the price itself, for existing prices changing currency or variant is confusing and disabled)
![solidus administration prices 2016-05-06 17-34-15](https://cloud.githubusercontent.com/assets/703401/15086750/e0f868b2-13b0-11e6-81b5-717f3be02106.png)

New view (all fields can be edited)
![solidus administration prices 2016-05-06 17-34-33](https://cloud.githubusercontent.com/assets/703401/15086751/e0fb4c1c-13b0-11e6-8f9a-7119c284140b.png)

gmacdougall commented on backend/spec/features/admin/products/pricing_spec.rb:
```diff
@@ -0,0 +1,63 @@
+        expect(page).to have_content(master_price.money.to_s)
+        expect(page).to have_content(master_price.currency)
+        expect(page).to have_content(other_price.money.to_s)
+        expect(page).to have_content(other_price.currency)
```
Can you please set expectations on the actual value (e.g. 'USD') instead?

mamhoff replied:
Sure, will do.

gmacdougall commented on backend/spec/controllers/spree/admin/prices_controller_spec.rb:
```diff
@@ -0,0 +1,38 @@
+      it 'assigns a search variable' do
+        subject
+        expect(assigns(:search)).to be_present
+        expect(assigns(:prices)).to be_present
```
I would like a better assertion here.

mamhoff commented:
@gmacdougall incorporated your feedback, and added pagination.

tvdeyen commented on backend/app/views/spree/admin/prices/_table.html.erb:
```diff
@@ -0,0 +1,53 @@
+        <% end %>
+        <% if can?(:destroy, price) %>
+          &nbsp;
+          <%#= link_to_delete(price, :no_text => true) unless price.deleted? %>
```
Is this intentional to disable the delete link here?

mamhoff replied:
Yes it is - there's no route for deletion yet. I should add that I guess :sheep: .

tvdeyen commented on backend/app/views/spree/admin/prices/_table.html.erb:
```diff
@@ -0,0 +1,53 @@
+
+  <tbody>
+  <% prices.each do |price| %>
+    <tr data-hook="prices_row" class="<%= "deleted" if price.deleted? %> <%= cycle('odd', 'even')%>">
```
Would be great if we could follow rails convention here and put this into a `_price` partial and render the `@prices` collection.

tvdeyen commented on core/config/locales/en.yml:
```diff
@@ -134,6 +134,10 @@ en:
+      spree/price:
+        currency: Currency
+        amount: Price
+        is_default: Currently Valid
```
"Currently valid" seems like a strange translation for `is_default` Boolean. I can't imagine what this does. "Currently valid" sounds like something date related. Do we plan to support time based pricing in the future?

mamhoff replied:
I would love to see time-based pricing. That aside, `is_default` is the most poorly named boolean I've seen so far: It's `true` be default, and adding a new price for a given currency will set all other prices for that currency/variant to `is_default: false`. So "Currently valid" is actually a pretty accurate translation for what it does.

jhawthorn commented:
:+1: tried this out. More work is needed, but I think it's good enough to merge. We can then collaborate on making it better.

Future steps:
* Validate entered currency (it's possible currently to make a price in an invalid currency, ex. "XYZ")
* Grouping common prices (@Dkendal and @Mandily did some investigation of this)
* We have the master price both on the main product page as well as the pricing page. This will probably be cause of some confusion.
* Showing "Master Variant" in the pricing table feels like we're leaking an implementation detail

tvdeyen commented:
I also think, this is good enough. How to we ensure to not lose track on the follow up tasks? Maybe a "pricing todo" issue?

jhawthorn commented:
@tvdeyen done. We can use #1167 as a meta-issue to track these

tvdeyen commented:
Thanks John.

mamhoff commented:
#1169 should fix the validation issue. I've amended this PR to use a `select2` instead of a `text` input to make it harder to input invalid data. As for the last point: I've removed the `.master_price` translation from this PR and switched to instead using `Variant#descriptive_name` (which has the same leakage issue, but is already there as model-level logic).

jhawthorn commented:
Thanks as always Martin
